qa/L0_storage_S3/test.sh

```diff
@@ -61,132 +61,169 @@ SERVER_TIMEOUT=360
 SERVER_LOG_BASE="./inference_server"
 source ../common/util.sh
 
-SERVER_LOG=$SERVER_LOG_BASE.log
-CLIENT_LOG=$CLIENT_LOG_BASE.log
+# First test with AWS ENV vars and then without AWS ENV vars
+for $ENV_VAR in "env" "no_env"; do
+    SERVER_LOG=$SERVER_LOG_BASE.$ENV_VAR.log
+    CLIENT_LOG=$CLIENT_LOG_BASE.$ENV_VAR.log
     
-rm -f $SERVER_LOG_BASE* $CLIENT_LOG_BASE*
+    if [ "$ENV_VAR" != "env" ]; then
+        unset AWS_ACCESS_KEY_ID
+        unset AWS_SECRET_ACCESS_KEY
+        unset AWS_DEFAULT_REGION
+    elif
+        aws configure set default.region "dummy_region" && \
```

Comment:
So the first time through the loop you set the configuration to be invalid... but that is ok because the envvars are going to be used.
The second time the envvars are unset... so now you ony have the configuration but isn't it invalid? And then below you set config again?
Can you run 3 iterations of the loop and at the start of each iteration setup envvars and config for each different case and clearly document what each iteration is testing. And then for the rest of the loop don't change env or config.

Reply:
The config needs to be set correctly so that the aws s3 cli command work. It is set to the correct creds and then set back to dummy creds after the s3 command is run.